### PR TITLE
Improved test coverage for Pluck & ForEach*

### PR DESCRIPTION
**Pull request by yknx4**
- [x] Pluck missing coverage
- [x] ForEach missing coverage

### CODE REVIEW

File: for_each_test.go [hunk 1]
```diff
@@ -55,3 +56,39 @@ func TestForEachContext(t *testing.T) {
 		require.Equal(t, testCase.Out, stringSlice)
 	}
 }
+
+func TestForEachContextError(t *testing.T) {
+	type contextKey string
```
Comment (cuducos) [inline for_each_test.go:61]: We have that declaration already in the package level (in `for_each_test.go` I guess) and, thus, we don't need to declare it again ; )

```suggestion
```

File: for_each_test.go [hunk 2]
```diff
@@ -55,3 +56,39 @@ func TestForEachContext(t *testing.T) {
...
+		return nil
+	}
+	ctx := context.WithValue(context.Background(), key, "a_value")
+
+	err := ForEachContextError(ctx, input, fn)
+	require.Nil(t, err)
+	require.Equal(t, input, calledInputs)
+}
+
+func TestForEachContextErrorWithError(t *testing.T) {
```
Comment (cuducos) [inline for_each_test.go:78]: We can use the `t.Run` and group both the success and error test cases under the `TestForEachContextError` ([example](https://github.com/Shopify/hoff/blob/6084ba63d8e84f98b0b70e081f7c0328fc0a33af/map_test.go#L31-L54)).

File: pluck_test.go [hunk 3]
```diff
@@ -12,6 +12,23 @@ func ExamplePluck() {
 	// Output: [[<nil>] [4]]
 }
 
+func TestPluckEmptyKeys(t *testing.T) {
+	output := Pluck(
+		[]map[string]any{
```
Comment (cuducos) [inline pluck_test.go:17]: Just out of curiosity: Is there any advantage of using `any` instead of `int` here?

Comment (yknx4) [reply pluck_test.go:17]: @cuducos not really afaik

**Review by chrispappas — APPROVED** [review]
nice

**Pull request merged**